pkg/gofr/datasource/pubsub/mqtt/mqtt.go

```diff
@@ -318,7 +319,15 @@ func getHandler(subscribeFunc SubscribeFunc) func(client mqtt.Client, msg mqtt.M
 		}
 
 		// call the user defined function
-		_ = subscribeFunc(pubsubMsg)
+		err := subscribeFunc(pubsubMsg)
```

Comment:
@vipul-rawat This seems to conflict with change I had to made due to code climate requirements (file was too long by 6 lines or so)
https://github.com/gofr-dev/gofr/pull/781/files#diff-01c8424bf0ca548fdc0070da34c818b40d70a65d64e08e10b7816bef1fea8f0e

Can we do this in single PR or so we don't lose this important fix in a merge?

Reply:
What do you suggest? In your PR #781 this function `SubscribeWithFunction` is already untouched. Shall we move this to `mqtt_sub.go`?

Reply:
Or I can move the extended functionalities of the MQTT client to a new file

Reply:
@vipul-rawat That huge PR size becoming pain in a lower back. Can you wait for it or commit right into that branch?

Reply:
I'll fix this after shutdown PR gets merged!